### ISSUE TITLE
[Crashlytics] Illegal instruction: 4 on Xcode Cloud during upload-symbols

### ISSUE THREAD

**Issue by skywalkerlw** [firebase/firebase-ios-sdk#11845]
### Description

 It was definitely working before with the same codebase and the same Xcode (14.3).

then around weeks ago, I could not run `upload-symbols` any more.

Here is the error message:

![image](https://github.com/firebase/firebase-ios-sdk/assets/2090084/ad4b6969-9149-4e76-b227-b2b8608168a0)

/Volumes/workspace/DerivedData/Build/Intermediates.noindex/ArchiveIntermediates/APAC_Prod/IntermediateBuildFilesPath/TravelAssist.build/Release-iphoneos/APAC_Prod.build/Script-279FBB7E289DFDD000526D82.sh: line 8:  8016 Illegal instruction: 4  "${PODS_ROOT}/FirebaseCrashlytics/upload-symbols" -gsp "${PROJECT_DIR}/Configuration/Firebase/APAC/prod/GoogleService-Info.plist" -p ios "${DWARF_DSYM_FOLDER_PATH}/${DWARF_DSYM_FILE_NAME}"

### Firebase SDK Version

10.11.0

### Xcode Version

14.2

### Installation Method

CocoaPods

### Firebase Product(s)

Crashlytics

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by max9631** [comment]
Same issue here

**Comment by rizafran** [comment]
Thanks for reporting, @skywalkerlw. I tried reproducing the issue on Xcode Cloud, but I'm unable to get the same error. Is it possible for you to share the script you used to upload your dSYMs from your shell script file (excluding any private information)?

**Comment by dneprDroid** [comment]
It reproduces on macOS Sonoma (Xcode Cloud)

**Comment by ansabalisf** [comment]
I am seeing the same issue as well.

**Comment by roby0970** [comment]
For the last few weeks, the script stopped working, but with no issues/errors reported. 

When running the script in Build Phase, which was working for the last 6 months, dSYMs stopped appearing in Firebase.
The command I have been running:
`${PODS_ROOT}/FirebaseCrashlytics/run -gsp ${PROJECT_DIR}/MyProject/GoogleService-Info.plist`
with input files: 
`${DWARF_DSYM_FOLDER_PATH}/${DWARF_DSYM_FILE_NAME}/Contents/Resources/DWARF/${TARGET_NAME}`.

After getting annoyed by manually uploading dSYMs, and as I have been using XCode Cloud, I tried to run the script in `ci_post_xcodebuild` step. But everything I tried would result in Illegal Instruction Exception which does not give me any useful information.

`$CI_PRIMARY_REPOSITORY_PATH/Pods/FirebaseCrashlytics/upload-symbols -d -gsp "$CI_PRIMARY_REPOSITORY_PATH/MyProject/GoogleService-Info.plist" -p ios "$CI_ARCHIVE_PATH/dSYMs"`

All paths are correct and files are existing in XCodeCloud env.

Local:
XCode 14.3.1
MacOS Ventura 13.4

XcodeCloud:
XCode 14.3.1
Latest (MacOS Sonoma 14 RC) 

Pod: Firebase/Crashlytics 10.3.0

Also, this has been happening on all of my projects built with XCodeCloud

**Comment by jaltreuter** [comment]
I started seeing this issue in Xcode Cloud recently too.  Here's a log from an example failure, which ran with Xcode 14.3.1 (14E300c) and macOS Sonoma 14 RC (23A339).

```
Run ci_post_xcodebuild.sh script
    Run command: 'cd /Volumes/workspace/repository/ci_scripts && CI_XCODEBUILD_EXIT_CODE=0 /Volumes/workspace/repository/ci_scripts/ci_post_xcodebuild.sh'
    Found valid archive path, trying to upload dSYMs.
    /Volumes/workspace/repository/ci_scripts/ci_post_xcodebuild.sh: line 8: 12494 Illegal instruction: 4  ../Pods/FirebaseCrashlytics/upload-symbols -gsp ../fastlane/$GSERVICE_PLIST_NAME -p ios $CI_ARCHIVE_PATH/dSYMs
    Command exited with non-zero exit-code: 132
```

Following the statements made earlier in this thread, I updated my Xcode Cloud build configuration to use macOS Ventura 13.5.2 (22G91) and the error went away.

This is my full `ci_post_xcodebuild.sh` script:

```
#!/bin/sh

set -e

if [[ -n $CI_ARCHIVE_PATH ]]; then
    echo "Found valid archive path, trying to upload dSYMs for config $GSERVICE_PLIST_NAME from $CI_ARCHIVE_PATH"
    ../Pods/FirebaseCrashlytics/upload-symbols -gsp ../fastlane/$GSERVICE_PLIST_NAME -p ios $CI_ARCHIVE_PATH/dSYMs
fi
```

**Comment by ansabalisf** [comment]
We resolved this problem by switching build os for Xcode cloud from MacOS 14 to MacOS 13. It still doesn't solve the actual problem but atleast builds are working fine now.

**Comment by roby0970** [comment]
Has anyone updated local MacOS to 14, to maybe check if combination of new Xcode and new MacOS is working? Or hopefully does a better error reporting in this case?

**Comment by syboMuhamed** [comment]
I'm testing issues that might pop up with macOS 14. I Ran 3 projects with iOS and Android builds Local:
      System Version: macOS 14.0 (23A344)
      Kernel-Version: Darwin 23.0.0
      Xcode 14.3.1.
The upload step fails on iOS part but works for Android.
 * - line 2:  6128 Illegal instruction: 4 - * 
 
 On baseline test with the same jobs/builds ran on macOS 13.6 and uploads worked fine.
 So I can confirm that something happens when upgrading macOS to 14.
 I ended up here trying to find a fix for the issue.

**Comment by kgrigsby59** [comment]
I am seeing the same issue as well.

**Comment by barakwei** [comment]
Same here, just ran `upload-symbols -gsp <path to path> -p ios <path to dsym zip file>` and it crashed on Sonoma (macOS 14.0). Works well on 13.6.

**Comment by syboMuhamed** [comment]
Installing Xcode 15 fixed the issue for me. 
The project setting was still set to use Xcode 14.3.1.
But installing Xcode 15 made uploading work.

**Comment by AliSoftware** [comment]
We're having the same issue on our CI (that runs on Buildkite, using our own Intel MacMini that run our build jobs in a Parallels VM) since we migrated our VMs to Xcode 15.

```
builder@xcode-15 tumblr-ios % /usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/Pods/FirebaseCrashlytics/upload-symbols -d -gsp /usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/Resources/GoogleService-Info.plist -p ios /usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/artifacts/publish/com.tumblr.Tumblr_319009_beta.app.dSYM.zip
upload-symbols 3.16 build 35

Arguments:
Debug mode enabled
Google App ID: {redacted}
Platform: ios
DSYM Paths: ["/usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/artifacts/publish/com.tumblr.Tumblr_319009_beta.app.dSYM.zip"]

Inspecting next path: /usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/artifacts/publish/com.tumblr.Tumblr_319009_beta.app.dSYM.zip
Found archive at path: /usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/artifacts/publish/com.tumblr.Tumblr_319009_beta.app.dSYM.zip
Inspecting next path: /var/folders/xl/csng4q450h96m4tbz0p12_xh0000gn/T///usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/artifacts/publish/com.tumblr.Tumblr_319009_beta.app.dSYM.zip.unzipped/Notification.appex.dSYM
Found dSYM at path: /var/folders/xl/csng4q450h96m4tbz0p12_xh0000gn/T///usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/artifacts/publish/com.tumblr.Tumblr_319009_beta.app.dSYM.zip.unzipped/Notification.appex.dSYM
Begin processing dSYM at /var/folders/xl/csng4q450h96m4tbz0p12_xh0000gn/T///usr/local/var/buildkite-agent/builds/MV-MKE-X64-012/automattic/tumblr-ios/artifacts/publish/com.tumblr.Tumblr_319009_beta.app.dSYM.zip.unzipped/Notification.appex.dSYM
zsh: illegal hardware instruction   -d -gsp  -p ios 
```

`xcodebuild -version` = `Xcode 15.0, Build version 15A240d`

**Comment by themiswang** [comment]
Upgraded my laptop to Sonoma but so far I could not reproduce the problem locally. Tested build on xcode15 through application and xcode14.3.1 through command line. They all work for me on Sonoma. I am a bit curious if this is an issue on x86 arch machine. I am on `23.0.0 Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:43 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T6000 arm64`

**Comment by barakwei** [comment]
This seems to be resolved on macOS 14.1
